sources/xfscli/src/bin/xfs_admin.rs

```diff
@@ -0,0 +1,239 @@
+            || self.lazy_counters.is_some()
+            || self.v5_feature.is_some()
+            || self.realtime_device.is_some()
+    }
+}
+
+fn main() {
+    let args: Args = argh::from_env();
+    match run(args) {
+        Ok(exit_code) => exit(exit_code),
+        Err(e) => {
+            eprintln!("{}", e);
+            exit(e.exit_code());
+        }
+    }
+}
+
+fn run(args: Args) -> Result<i32> {
+    // Return the version for arg version as true
+    if args.version {
+        return Ok(query_xfs_db(vec!["-p", "xfs_admin", "-V"]));
+    }
+
+    // Collect all the arguments for XFS_DB, XFS_IO and XFS_REPAIR commands
+    // by parsing the CLI inputs.
+    let mut db_opts: Vec<&str> = vec![];
+    let mut io_opts: Vec<&str> = vec![];
+    let mut repair_opts: Vec<&str> = vec![];
+
+    let mut lazycount_arg = "lazycount=".to_string();
+    if let Some(lazy_counters) = &args.lazy_counters {
+        lazycount_arg.push_str(lazy_counters.to_string().as_str());
+        repair_opts.append(&mut vec!["-c", lazycount_arg.as_str()]);
+    }
+
+    if args.extent_flag {
+        db_opts.append(&mut vec!["-c", "version extflg"]);
+    }
+
+    if args.filesystem_image {
+        db_opts.push("-f");
+    }
+
+    if args.log_version_2 {
+        db_opts.append(&mut vec!["-c", "version log2"]);
+    }
+
+    if args.filesystem_label {
+        db_opts.append(&mut vec!["-r", "-c", "label"]);
+        io_opts.append(&mut vec!["-r", "-c", "label"]);
+    }
+
+    let mut label_arg_db = "label ".to_string();
+    let mut label_arg_io = "label ".to_string();
+    if let Some(label) = &args.label {
+        label_arg_db.push_str(label.as_str());
+        db_opts.append(&mut vec!["-c", label_arg_db.as_str()]);
+
+        match label.as_str() {
+            "--" => label_arg_io.push_str(label.as_str()),
+            _ => {
+                label_arg_io.push_str("-s ");
+                label_arg_io.push_str(label.as_str())
+            }
+        }
+        io_opts.append(&mut vec!["-c", label_arg_io.as_str()]);
+    }
+
+    if let Some(v5_feature) = &args.v5_feature {
+        repair_opts.append(&mut vec!["-c", v5_feature.as_str()]);
+    }
+
+    if args.project_identifier_32_bit {
+        db_opts.append(&mut vec!["-c", "version projid32bit"]);
+    }
+
+    if args.filesystem_uuid {
+        db_opts.append(&mut vec!["-r", "-c", "uuid"]);
+        io_opts.append(&mut vec!["-r", "-c", "fsuuid"]);
+    }
+
+    let mut uuid_arg = "uuid ".to_string();
+    if let Some(uuid) = &args.uuid {
+        uuid_arg.push_str(uuid.as_str());
+        db_opts.append(&mut vec!["-c", uuid_arg.as_str()]);
+    }
+
+    let device = &args.device.as_ref().context(error::ParseTargetSnafu)?;
+
+    // Exit if filesystem is mounted and require offline for queried arguments
+    if let Some(mpt) = get_mount_point(device) {
+        // filesystem is mounted
+        if args.is_required_offline() {
+            return Err(Error::MountedFilesystem { mount_point: mpt });
```

Comment:
I expected this to be `MountedFilesystemSnafu`?

Reply:
I think Snafu is needed when we use context?